fedimg/services/ec2/ec2initiate.py

```diff
@@ -163,5 +163,8 @@ def main(image_urls, access_id, secret_key, regions, volume_types=None,
             #TODO: Implement the clean up of the images if failed.
             # uploader.clean_up(image_id=image.id, delete_snapshot=True)
 
-    shutil.rmtree(os.path.dirname(source))
+        if source:
```

Comment:
in what cases does source evaluate to non true? I don't see what problem this is solving.

Reply:
There could be various reasons when source evaluates to non true, euca2ools does not setup, or wget fails.
fedimg needs to be better at raising and catching exceptions. I have opened up #113 for the same.

Reply:
it looks like this should be in a `finally` block then. If the valueError gets raised above then this code wouldn't get hit right?